python/langsmith/client.py

```diff
@@ -931,6 +933,24 @@ def request_with_retries(
                     filler = "*" * (max(0, len(api_key) - 7))
                     masked_api_key = f"{prefix}{filler}{suffix}"
 
+                    debug_crash_dump_file = ls_utils.get_env_var(
+                        "LANGSMITH_DEBUG_CRASH_DUMP"
+                    )
+                    if debug_crash_dump_file is not None and e.request is not None:
+                        try:
+                            request_data: (
+                                requests.Request | requests.PreparedRequest
+                            ) = copy.deepcopy(e.request)
```

Comment:
We should avoid deep copies. Especially just so we can put a masked API key in the headers